conda_smithy/templates/run_docker_build.sh.tmpl

```diff
@@ -39,6 +39,27 @@ mkdir -p "$ARTIFACTS"
 DONE_CANARY="$ARTIFACTS/conda-forge-build-done-${CONFIG}"
 rm -f "$DONE_CANARY"
 
+# Set up the docker image that we are going to use.  This is used to install yum requirements at build time if needed
+{{ docker.executable }} pull $DOCKER_IMAGE
+
+# If we have yum requirements build a new specialized image that we will use to run our build in that includes
+# those dependencies.
+pushd ${RECIPE_ROOT}
+if [ -f ${RECIPE_ROOT}/yum_requirements.txt ]; then
+
+    cat >Dockerfile.yum <<DOCKERFILE
+FROM ${DOCKER_IMAGE}
+ADD yum_requirements.txt /opt/requirements.txt
+RUN yum -y install \$(cat /opt/requirements.txt)
+DOCKERFILE
+
+    export NEW_DOCKER_IMAGE="conda-smithy-$(openssl rand -hex 12)"
```

Comment:
Can the dependency on openssl be dropped? I'd like to have `run_docker_build.sh` working without additional dependencies.

Reply:
since this is used with git, you will have openssl around?

Reply:
but we can change this to antyhing that emits a randomish number, this was just the easiest way

Reply:
Why not do this inside the docker container? There you can have openssl installed via conda

Reply:
we need the name of the docker image in order to run it....

Reply:
Sorry for being dense, why do we need a random number here?

Reply:
FWIW I also share @isuruf's concerns with increasing reliance on what is outside the Docker image. Raised issue ( https://github.com/conda-forge/conda-smithy/issues/1020 ) to discuss cutting back on existing things that are required outside the image to make builds easier to reproduce and debug locally.